### ISSUE TITLE
Support NotNullIfNotNull on operator

### ISSUE THREAD

**Issue by igor-tkachev** [dotnet/roslyn#37870]
**16.2.1**: 

**Steps to Reproduce**:

```c#
class Program
{
	class A { }
	class B
	{
		[return: NotNullIfNotNull("b")]
		public static implicit operator A?(B? b) => b == null ? null : new A();
	}

	static void Main(string[] args)
	{
		B b = new B();
		A a = b;
	}
}
```
[sharplab](https://sharplab.io/#v2:D4AQDABCCMB0AiBLAhgcwHYHsDOAXRAxtrAMKYAmApgILrIA2AntotgNwCwAUNwMToBXevWQAjepQiU64ytxAAmKNADs3AN7cAkIojUI6iAF9tugELbNXLVoDaIFQC4IAOUy4XQ+gEkAZm48vAAoAIlEQgEoAXW0dAGZlADYIRABbAAd6QkRcCEx0ygAnZFxMQr0AfiCzCohRCIgAXgA+OqbGiEFhCFqu+ghndEoAdz0giM5rEx5rGGSQABYIAFlkRHQgmDBbKIhkQtRsCMtYszaOodGzccmbfWQmurYIAHoXiGH99HXUCBIAZQAHIkwGBnGR0AA3Ir4dC/PoQLK4IoMPLldI4FiyTpeCCQhgCSSlTqYdAAWj6YgkEFwjAKsG00yMQA=)

**Expected Behavior**:

This code should not generate any warnings.

**Actual Behavior**:

```
warning CS8600: Converting null literal or possible null value to non-nullable type.
```

**Comment by dlech** [comment]
Duplicate of  #32671?

**Comment by jcouv** [comment]
Relates to https://github.com/dotnet/roslyn/issues/40113 (Cast with user-defined conversion doesn't honor [AllowNull])

**Comment by jcouv** [comment]
Actually, @dlech is correct and both should be marked as duplicates of https://github.com/dotnet/roslyn/issues/32671 (Annotations do not work for user defined conversion operators)

I'll go ahead and close this issue. Thanks
